proofs/hol_light/x86_64/README.md

```diff
@@ -0,0 +1,99 @@
+    - zetas: NTT constants pointer
+    - x: Original polynomial coefficients
+    - pc: Current value of Program Counter (PC)
+    - stackpointer: Stack pointer
+    - returnaddress: Return address on the stack *)
+`!a zetas x pc stackpointer returnaddress.
+    (* Alignment and non-overlapping requirements *)
+    aligned 32 a /\
+    aligned 32 zetas /\
+    nonoverlapping (word pc,LENGTH mldsa_ntt_mc) (a, 1024) /\
+    nonoverlapping (word pc,LENGTH mldsa_ntt_mc) (zetas, 2496) /\
+    nonoverlapping (a, 1024) (zetas, 2496) /\
+    nonoverlapping (stackpointer,8) (a, 1024) /\
+    nonoverlapping (stackpointer,8) (zetas, 2496)
+    ==> ensures x86
+      (* Precondition *)
+      (\s. (* The memory at the current PC is the byte-code of mldsa_ntt() *)
+        bytes_loaded s (word pc) mldsa_ntt_mc /\
+        read RIP s = word pc /\
+        read RSP s = stackpointer /\
+        (* The return address is on the stack *)
+        read (memory :> bytes64 stackpointer) s = returnaddress /\
+        (* Arguments are passed via C calling convention *)
+        C_ARGUMENTS [a; zetas] s /\
+        (* NTT constants are properly loaded *)
+        wordlist_from_memory(zetas,624) s = MAP (iword: int -> 32 word) mldsa_complete_qdata /\
+        (* Input bounds checking *)
+        (!i. i < 256 ==> abs(ival(x i)) <= &8380416) /\
+        (* Give a name to the memory contents at the source pointer *)
+        !i. i < 256
+            ==> read(memory :> bytes32(word_add a (word(4 * i)))) s = x i)
+      (* Postcondition: Eventually we reach a state where ... *)
+      (\s.
+        (* The PC is the return address *)
+        read RIP s = returnaddress /\
+        (* Stack pointer is adjusted *)
+        read RSP s = word_add stackpointer (word 8) /\
+        (* The integers represented by the final memory contents
+         * are congruent to the ML-DSA forward NTT transformation
+         * of the original coefficients, modulo 8380417, with proper bounds *)
+        !i. i < 256
+            ==> let zi = read(memory :> bytes32(word_add a (word(4 * i)))) s in
+                (ival zi == mldsa_forward_ntt (ival o x) i) (mod &8380417) /\
+                abs(ival zi) <= &42035261)
+      (* Footprint: The program may modify (only) the ABI permitted registers
+       * and flags, stack pointer, and the memory contents at the source pointer. *)
+      (MAYCHANGE [RSP] ,, MAYCHANGE_REGS_AND_FLAGS_PERMITTED_BY_ABI ,,
+       MAYCHANGE [memory :> bytes(a,1024)])`
+```
+
+## Reproducing the proofs
+
+To reproduce the proofs, enter the nix shell via
+
+```bash
+nix develop --experimental-features 'nix-command flakes'
+```
+
+from mldsa-native's base directory. Then
+
+```bash
+make -C proofs/hol_light/x86_64
+```
+
+will build and run the proofs. Note that this make take hours even on powerful machines.
+
+For convenience, you can also use `tests hol_light` which wraps the `make` invocation above; see `tests hol_light --help`.
+
```

Comment:
Can we document on which platforms this is expected to work on? Linux (x86/AArch64) and x86 Mac? The notably exception, Arm Mac, should ideally be mentioned as well, with a reference to the issue describing the problem.